src/components/workexperience/WorkExperienceShown.js

```diff
@@ -11,24 +17,58 @@ const WorkExperienceShown = ({
   achievements,
   refereeName,
   refereeContact,
+  currentlyWorking,
   preview,
   onHiddenIconClicked,
+  onDelete,
+  onContactLinkClicked,
 }) => {
   const [hidden, setHidden] = useState(false);
-
-  const rolesList = roles.split('.').map(role => <li key={role}>{role}</li>);
-  const achievementsList = achievements.split('.').map(achievement => <li key={achievement}>{achievement}</li>);
+  const [editWork, setEdit] = useState(false);
 
   if (hidden && preview) {
     return <></>;
   }
 
-  const onHiddenIconClickedHandler = e => {
-    e.preventDefault();
+  const rolesList = roles
+    .split('.')
+    .filter(role => {
+      if (role.trim()) {
+        return role;
+      }
+    })
+    .map((role, index) => <li key={index}>{role}</li>);
+
+  let achievementsList = null;
+
+  if (achievements.length > 1) {
+    achievementsList = achievements
+      .split('.')
+      .filter(achievement => {
+        if (achievement.trim()) {
+          return achievement;
+        }
+      })
+      .map((achievement, index) => <li key={index}>{achievement}</li>);
+  }
+
+  let labelForDifference = dateUtils.getDifferenceYearMonth(startDate, endDate, currentlyWorking);
+
+  labelForDifference = labelForDifference ? `( ${labelForDifference} )` : '';
+
+  const refereeDetail = /^\d+$/.test(refereeContact) ? ` ${COUNTRY_CODE} - ${refereeContact}` : ` ${refereeContact}`;
```

Comment:
`/^\d+$/` should be in the regex constant file. right?

Reply:
yes, I have moved it.